lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/StackPresenter.java

```diff
@@ -485,16 +498,18 @@ private void mergeBackButton(TopBarButtons buttons, StackController stack) {
         return result;
     }
 
-    private void mergeTopBarOptions(TopBarOptions resolveOptions, Options options, StackController stack, ViewController<?> child) {
-        TopBarOptions topBarOptions = options.topBar;
+    private void mergeTopBarOptions(TopBarOptions resolveOptions, Options toMerge, StackController stack,
+                                    ViewController<?> child) {
+        TopBarOptions topBarOptions = toMerge.topBar;
         final View component = child.getView();
-        if (options.layout.direction.hasValue()) topBar.setLayoutDirection(options.layout.direction);
+        if (toMerge.layout.direction.hasValue()) topBar.setLayoutDirection(toMerge.layout.direction);
         if (topBarOptions.height.hasValue()) topBar.setHeight(topBarOptions.height.get());
         if (topBarOptions.elevation.hasValue()) topBar.setElevation(topBarOptions.elevation.get());
         if (topBarOptions.topMargin.hasValue() && topBar.getLayoutParams() instanceof MarginLayoutParams) {
             ((MarginLayoutParams) topBar.getLayoutParams()).topMargin = UiUtils.dpToPx(activity, topBarOptions.topMargin.get());
         }
-        mergeStatusBarDrawBehindOptions(resolveOptions, options);
+        Options childOptions = stack.resolveChildOptions(child).mergeWith(toMerge).withDefaultOptions(defaultOptions);
+        mergeStatusBarDrawBehindOptions(resolveOptions, childOptions);
```

Comment:
this one as well

Reply:
Since topBar position depends on status bar options we need to merge those options and have them every time we need to update the topBar position we need the current options state of the status bar, taking into consideration the child options and the options that passed to `mergeOptions` and default options, this was done in the calling function `mergeTopBarOptions` but it's done for the topBar options only, and changing the calling function arguments would cause another changes across the files and the tests.

think we might leave it as is, and the next PR that deals with options aggregation will have proper change.